lib/Echidna/ABI.hs

```diff
@@ -210,13 +209,8 @@ canShrinkAbiValue (AbiArrayDynamic _ l) = l /= mempty
 canShrinkAbiValue (AbiTuple v)          = any canShrinkAbiValue v
 canShrinkAbiValue _                     = True
 
-bounds :: forall a. (Bounded a, Integral a) => a -> (Integer, Integer)
-bounds = const (fromIntegral (0 :: a), fromIntegral (maxBound :: a))
-
--- | Given a bitvector representation of an integer type, randomly change bits, shrinking it towards 0
-shrinkInt :: forall a m. (Bits a, Bounded a, Integral a, MonadRandom m) => a -> m a
-shrinkInt x | x == -1   = pure 0
-            | otherwise = (if x < 0 then (.|.) else (.&.)) x . fromIntegral <$> getRandomR (bounds x)
+shrinkInt :: (Integral a, MonadRandom m) => a -> m a
+shrinkInt x = fromIntegral <$> getRandomR (0, toInteger x)
```

Comment:
maybe we also want to keep the "bitwise shrinking". i kinda like the idea of being able to switch off bits.

Reply:
I can bring it back if there is a good use case for it. Not sure if it is worth complicating the code though. The random smaller number turned out to be simpler and more effective strategy.